Dockerfile

```diff
@@ -1,56 +1,53 @@
 ARG package=arcaflow_plugin_uperf
 
-# build poetry
-FROM quay.io/centos/centos:stream8 as poetry
+# STAGE 1 -- Build module dependencies and run tests
+# The 'poetry' and 'coverage' modules are installed and verson-controlled in the
+# quay.io/arcalot/arcaflow-plugin-baseimage-python-buildbase image to limit drift
+FROM quay.io/arcalot/arcaflow-plugin-baseimage-python-buildbase:0.2.0 as build
 ARG package
 RUN dnf module -y install python39 && dnf install --setopt=tsflags=nodocs -y python39 python39-pip lksctp-tools-devel && dnf clean all \
  && dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm \
  && dnf -y install uperf
 
-WORKDIR /app
-
 COPY poetry.lock /app/
 COPY pyproject.toml /app/
 
-RUN python3.9 -m pip install poetry==1.4.2 \
- && python3.9 -m poetry config virtualenvs.create false \
- && python3.9 -m poetry install --without dev --no-root \
- && python3.9 -m poetry export -f requirements.txt --output requirements.txt --without-hashes
+# Convert the dependencies from poetry to a static requirements.txt file
+RUN python -m poetry install --without dev --no-root \
+ && python -m poetry export -f requirements.txt --output requirements.txt --without-hashes
 
-# run tests
 COPY ${package}/ /app/${package}
-COPY tests /app/tests
+COPY tests /app/${package}/tests
 
 ENV PYTHONPATH /app/${package}
+WORKDIR /app/${package}
 
-RUN mkdir /htmlcov \
- && pip3 install coverage \
- && python3 -m coverage run tests/test_uperf_plugin.py \
- && python3 -m coverage html -d /htmlcov --omit=/usr/local/*
+# Run tests and return coverage analysis
+RUN python -m coverage run tests/test_${package}.py \
+ && python -m coverage html -d /htmlcov --omit=/usr/local/*
 
 
-# final image
-FROM quay.io/centos/centos:stream8
+# STAGE 2 -- Build final plugin image
+FROM quay.io/arcalot/arcaflow-plugin-baseimage-python-osbase:0.2.0
 ARG package
 RUN dnf module -y install python39 && dnf install --setopt=tsflags=nodocs -y python39 python39-pip lksctp-tools-devel && dnf clean all \
```

Comment:
Installation of python module and packages should be removed

Reply:
Removed